### PR TITLE
Add brightScriptConsolePort option. Use remotePort more

### PR DESCRIPTION
**Pull request by TwitchBronBron**
- Use the `remotePort` launch configuration option when sending home presses
- refactor the adapters to use a common configuration object parameter instead of ordered params.
- add `brightScriptConsolePort` to override where telnet connects.

### CODE REVIEW

File: src/LaunchConfiguration.ts [hunk 1]
```diff
@@ -160,6 +160,11 @@ export interface LaunchConfiguration extends DebugProtocol.LaunchRequestArgument
      */
     remotePort?: number;
 
+    /**
+     * The brightscript console port. In telnet mode this is the port used for the telnet connection. In debug protocol mode, this is where the compile errors show up.
```
Comment (chrisdp) [inline src/LaunchConfiguration.ts:164]:
```suggestion
     * The brightscript console port. In telnet mode this is the port used for the telnet connection. In debug protocol mode, this is used to obtain compile errors from the device.
```

File: src/util.ts [hunk 2]
```diff
@@ -381,6 +382,11 @@ class Util {
         const seconds = allocate(1000);
         return `${hours > 0 ? hours + 'h' : ''}${minutes > 0 ? minutes + 'm' : ''}${seconds > 0 ? seconds + 's' : ''}${ms > 0 ? ms + 'ms' : ''}`;
     }
+
+    public normalizeAdapterOptions(options: AdapterOptions) {
```
Comment (chrisdp) [inline src/util.ts:386]: we can't default an ip but is there anything we should do here? Like throw an error? I know we "should" catch it earlier but wouldn't be bad to do here would it?

File: src/adapters/TelnetAdapter.ts [hunk 3]
```diff
@@ -232,12 +236,12 @@ export class TelnetAdapter {
...
             //if the connection fails, reject the connect promise
             client.addListener('error', (err) => {
-                deferred.reject(new Error(`Error with connection to: ${this.host} \n\n ${err.message} `));
+                deferred.reject(new Error(`Error with connection to: ${this.options.host} \n\n ${err.message} `));
             });
 
             const settlePromise = this.settle(client, 'data');
-            client.connect(8085, this.host, () => {
-                this.logger.log(`Telnet connection established to ${this.host}`);
+            client.connect(8085, this.options.host, () => {
```
Comment (chrisdp) [inline src/adapters/TelnetAdapter.ts:243]:
```suggestion
            client.connect(this.options.brightScriptConsolePort, this.options.host, () => {
```

File: src/util.ts (hunk 2 above)
Comment (TwitchBronBron) [reply src/util.ts:386]: Nah. This is mostly just to centralize the default ports. If we want to catch a missing host, that should be done at the vscode level or in the initialize function of the session.

File: src/adapters/TelnetAdapter.ts [hunk 4]
```diff
@@ -232,12 +236,12 @@ export class TelnetAdapter {
 
             //if the connection fails, reject the connect promise
             client.addListener('error', (err) => {
-                deferred.reject(new Error(`Error with connection to: ${this.host} \n\n ${err.message} `));
+                deferred.reject(new Error(`Error with connection to: ${this.options.host} \n\n ${err.message} `));
```
Comment (chrisdp) [inline src/adapters/TelnetAdapter.ts:239]:
```suggestion
                deferred.reject(new Error(`Error with connection to: ${this.options.host}:${this.options.brightScriptConsolePort} \n\n ${err.message} `));
```

File: src/adapters/TelnetAdapter.ts [hunk 5]
```diff
@@ -232,12 +236,12 @@ export class TelnetAdapter {
...
             });
 
             const settlePromise = this.settle(client, 'data');
-            client.connect(8085, this.host, () => {
-                this.logger.log(`Telnet connection established to ${this.host}`);
+            client.connect(8085, this.options.host, () => {
+                this.logger.log(`Telnet connection established to ${this.options.host}`);
```
Comment (chrisdp) [inline src/adapters/TelnetAdapter.ts:244]:
```suggestion
                this.logger.log(`Telnet connection established to ${this.options.host}:${this.options.brightScriptConsolePort}`);
```

File: src/adapters/DebugProtocolAdapter.ts [hunk 6]
```diff
@@ -298,11 +295,11 @@ export class DebugProtocolAdapter {
 
             //if the connection fails, reject the connect promise
             this.compileClient.addListener('error', (err) => {
-                deferred.reject(new Error(`Error with connection to: ${this.host} \n\n ${err.message} `));
+                deferred.reject(new Error(`Error with connection to: ${this.options.host} \n\n ${err.message} `));
```
Comment (chrisdp) [inline src/adapters/DebugProtocolAdapter.ts:298]:
```suggestion
                deferred.reject(new Error(`Error with connection to: ${this.options.host}:${this.options.brightScriptConsolePort} \n\n ${err.message} `));
```

File: src/adapters/DebugProtocolAdapter.ts [hunk 7]
```diff
@@ -298,11 +295,11 @@ export class DebugProtocolAdapter {
...
             });
-            this.logger.info('Connecting via telnet to gether compile info', { host: this.host });
-            this.compileClient.connect(8085, this.host, () => {
-                this.logger.log(`Connected via telnet to gather compile info`, { host: this.host });
+            this.logger.info('Connecting via telnet to gether compile info', { host: this.options.host });
```
Comment (chrisdp) [inline src/adapters/DebugProtocolAdapter.ts:300]:
```suggestion
            this.logger.info('Connecting via telnet to gather compile info', { host: this.options.host, port: this.options.brightScriptConsolePort });
```

File: src/adapters/DebugProtocolAdapter.ts [hunk 8]
```diff
@@ -298,11 +295,11 @@ export class DebugProtocolAdapter {
...
+            this.compileClient.connect(this.options.brightScriptConsolePort, this.options.host, () => {
+                this.logger.log(`Connected via telnet to gather compile info`, { host: this.options.host });
```
Comment (chrisdp) [inline src/adapters/DebugProtocolAdapter.ts:302]:
```suggestion
                this.logger.log(`Connected via telnet to gather compile info`, { host: this.options.host, port: this.options.brightScriptConsolePort });
```

**Review by chrisdp — APPROVED** [review]

**Pull request merged**